### PR TITLE
Color provider improvements

### LINKED ISSUE #2541: BlockModelRenderer.field_32784 -> MAX_BRIGHTNESS_CACHE_SIZE
Used for the brightness cache's maps. The code checks that they never go past `field_32784`'s value (100).

### LINKED ISSUE #2542: BlockColors.field_32163, ItemColors.field_32165 -> NO_COLOR
Used as the return value of the default `BlockColorProvider`s and `ItemColorProvider`s when there's no colour given for parameters. (Ignored tint index, no world etc)

### LINKED ISSUE #2550: BlockColors.getColor(BlockState, World, BlockPos) -> getParticleColor
It's used exclusive for particles and unlike the other `getColor`, falls back to the map color of the block state. It also forces the tint index 0 (used for particles).

The other method, `getColor(BlockState, BlockRenderView, BlockPos, int)`, has a customisable `tintIndex` parameter and is used for actually fetching the rendering color of a block.

### PR DESCRIPTION
**Pull request by Juuxel**
This PR focuses on `Block/ItemColorProvider` and `Block/ItemColors`.

- `BlockColors.getColor(BlockState, World, BlockPos) -> getParticleColor` (closes #2550)
  - Only used for particles; includes a fallback to map color, so not a pure color provider method. In addition, it hardcodes the tint index 0, which makes it unusable for most other stuff.
- `BlockModelRenderer.colorMap`, `ItemRenderer.colorMap` -> `colors`.
  - To follow the class names `BlockColors` and `ItemColors`.
  - Follow-up for #1372.
- `ItemColors.getColorMultiplier` -> `getColor`
  - To match the method names in `BlockColors` and both color provider interfaces.
- Added javadoc for some methods.
- Mapped constants
  - `BlockColors.field_32163`, `ItemColors.field_32165` -> `NO_COLOR` (closes #2542)
    - The value used when there's no block/item color for a tint index.

### CODE REVIEW

File: mappings/net/minecraft/client/color/block/BlockColorProvider.mapping [hunk 1]
```diff
@@ -1,5 +1,7 @@
 CLASS net/minecraft/class_322 net/minecraft/client/color/block/BlockColorProvider
 	METHOD getColor (Lnet/minecraft/class_2680;Lnet/minecraft/class_1920;Lnet/minecraft/class_2338;I)I
+		COMMENT {@return the color of the block state for the specified tint index,
+		COMMENT or -1 if not tinted}
```
Comment (liach) [inline mappings/net/minecraft/client/color/block/BlockColorProvider.mapping:4]: hmm, is -1 just fully white color, so it's effectively not tinted?

**Review by liach — APPROVED** [review]

File: mappings/net/minecraft/client/color/block/BlockColorProvider.mapping (hunk 1 above)
Comment (Juuxel) [reply mappings/net/minecraft/client/color/block/BlockColorProvider.mapping:4]: Technically, yes. However, the actual constant is still probably -1, as using it defines the alpha value which is usually left out of the color provider return values. (The actual alpha is always ignored and set to 0xFF)

Comment (liach) [reply mappings/net/minecraft/client/color/block/BlockColorProvider.mapping:4]: I guess they must have this as a constant somewhere... Hope we can unpick it at some point in the future

Comment (Juuxel) [reply mappings/net/minecraft/client/color/block/BlockColorProvider.mapping:4]: It's the private constant that I mapped in BlockColors and ItemColors. Couldn't get it to unpick, though - maybe some lambda issues?

Comment (liach) [reply mappings/net/minecraft/client/color/block/BlockColorProvider.mapping:4]: Hmm yes, lambda methods don't explicitly override the single abstract method in the interface, and detection does need to check each bootstrap method (or indy callsite) manually, and I doubt if unpick even has that check

**Review by YanisBft — APPROVED** [review]

**Comment by modmuss50** [comment]
Sorry, seems there are merge conflicts. Thanks

**Comment by Juuxel** [comment]
Fixed the merge conflicts now.

**Pull request merged**